Shoko.Server/Import/Importer.cs

```diff
@@ -1438,49 +1438,41 @@ public static void CheckForTraktTokenUpdate(bool forceRefresh)
         try
         {
             var settings = Utils.SettingsProvider.GetSettings();
-            if (!settings.TraktTv.Enabled)
+            if (!settings.TraktTv.Enabled || string.IsNullOrEmpty(settings.TraktTv.TokenExpirationDate))
             {
                 return;
             }
 
             var traktHelper = Utils.ServiceContainer.GetRequiredService<TraktTVHelper>();
-            // by updating the Trakt token regularly, the user won't need to authorize again
-            var freqHours = 24; // we need to update this daily
 
-            var sched =
-                RepoFactory.ScheduledUpdate.GetByUpdateType((int)ScheduledUpdateType.TraktToken);
-            if (sched != null)
+            // Convert the Unix timestamp to DateTime directly
+            var expirationDate = DateTimeOffset.FromUnixTimeSeconds(long.Parse(settings.TraktTv.TokenExpirationDate)).DateTime;
+
+            // Check if force refresh is requested or the token is expired
+            if (forceRefresh || DateTime.Now.Add(TimeSpan.FromDays(45)) >= expirationDate)
```

Comment:
Why 45 days?

Reply:
https://discord.com/channels/96234011612958720/1148092356592160768/1195127627397214331

Initially I had it set to 30 days, but half would be 45 days so I set this. We can set it to whatever is fine

Reply:
Half of what

Reply:
Fair enough. Looks good then